services/worker-manager/src/providers/aws.js

```diff
@@ -280,13 +310,15 @@ class AwsProvider extends Provider {
     });
   }
 
-  // should this be implemented on Provider? Looks like it's going to be the same for all providers
   async scanPrepare() {
     this.seen = {};
-    this.errors = {};
   }
 
-  async scanCleanup() {
+  async scanCleanup({responsibleFor}) {
+    for (const workerPoolId of responsibleFor) {
+      this.seen[workerPoolId] = this.seen[workerPoolId] || 0;
```

Comment:
👍

do we also need `this.errors[workerPoolId] = this.errors[workerPoolId] || [];`?